src/stub/core_impl/thread_extra_info.rs

```diff
@@ -0,0 +1,37 @@
+use super::prelude::*;
+use crate::protocol::commands::ext::ThreadExtraInfo;
+use crate::protocol::SpecificIdKind;
+use crate::target::ext::base::BaseOps;
+
+impl<T: Target, C: Connection> GdbStubImpl<T, C> {
+    pub(crate) fn handle_thread_extra_info<'a>(
+        &mut self,
+        res: &mut ResponseWriter<'_, C>,
+        target: &mut T,
+        command: ThreadExtraInfo<'a>,
+    ) -> Result<HandlerStatus, Error<T::Error, C::Error>> {
+        let ops = match target.base_ops() {
+            BaseOps::SingleThread(_) => return Ok(HandlerStatus::Handled),
+            BaseOps::MultiThread(ops) => ops,
+        };
```

Comment:
so, this guard need to be modified to ensure that the rest of the function gets optimized out if `support_thread_extra_info` returns `None`.

The following should do what you want (along with simplifying some of the `ops` code down below as well, removing the need for that `PacketUnexpected` arm).

```suggestion
        let ops = match target.base_ops() {
            BaseOps::SingleThread(_) => return Ok(HandlerStatus::Handled),
            BaseOps::MultiThread(ops) => match ops.support_thread_extra_info() {
                Some(ops) => ops,
                None => return Ok(HandlerStatus::Handled),
            },
        };
```

And please add in the `__dead_code_marker` as present in other handlers, and validate that this bit of code does indeed get dead code eliminated in the case that the target doesn't implement this IDET. (i.e: by temporarily commenting it out in the armv4t example, and running the `test_dead_code_elim.sh` script). 

In addition, please make sure that running `./example_no_std/check_size.sh` returns the same value before / after this PR.

Reply:
Updated as requested.

`test_dead_code_elim.sh` shows the path is getting eliminated properly.  `check_size.sh` shows an increase of exactly 1 byte in .text.  I'm still trying to trace down where that is coming from.

Reply:
huh, interesting. a single byte difference? that's weird...

well, in any case, I wouldn't worry too much about that 😄

Reply:
Looks like the 1 byte difference is just the optimizer reordering some jmp / calls such that we end up with a slightly different generated main code.  None of the calls in question are in the new IDET paths, but it's enough to cause the compiler to generate code slightly differently.

I can provide assembly diffs if interested but my read is that this may be very difficult to avoid.

Reply:
yep, that's what I would've guessed. I wouldn't sweat it - different rustc versions introduce plenty more variation already. the key thing is that we aren't seeing massive bumps in size (i.e: hundreds of bytes).